proxy-node-gateway/src/main/java/uk/gov/ida/notification/RemoveSessionCookieFilter.java

```diff
@@ -0,0 +1,44 @@
+package uk.gov.ida.notification;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Arrays;
+
+import static uk.gov.ida.notification.session.SessionCookieService.COOKIE_X_PN_SESSION;
+
+public class RemoveSessionCookieFilter implements Filter {
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain chain) throws IOException, ServletException {
+        chain.doFilter(servletRequest, servletResponse);
+        HttpServletRequest request = (HttpServletRequest) servletRequest;
+        Cookie[] cookies = request.getCookies();
+        if (cookies != null) {
+            Arrays.stream(cookies)
+                    .filter(c -> COOKIE_X_PN_SESSION.equals(c.getName()))
+                    .findFirst()
```

Comment:
It's possible for there to be more than one cookie with the same name. I'm not sure if we'd ever get into that scenario but it might be worth invalidating anything with the right name and not just the first in the stream.